### PR TITLE
fix: handle non-2xx GitHub API responses in release script (fixes #6566)

### LINKED ISSUE #6566: Race condition in release workflow causes failure in artifacts publish
## Describe the bug

https://statics.teams.cdn.office.net/evergreen-assets/safelinks/2/atp-safelinks.html

release script seems to be trying to delete artifacts which are not in the release yet

```
Run # Parse repository to get owner and repo names
Namespace(owner='spiceai', repo='spiceai', tag='v1.5.0', release_name=None, body=None, prerelease=None, action='delete', artifact=['./release/spice_linux_x86_64.tar.gz', './release/spiced_linux_x86_64.tar.gz', './release/spiced_models_linux_x86_64.tar.gz'])
Deleting -> spice_linux_x86_64.tar.gz
Traceback (most recent call last):
  File "/home/runner/work/spiceai/spiceai/./.github/scripts/github_release.py", line 160, in <module>
    main()
  File "/home/runner/work/spiceai/spiceai/./.github/scripts/github_release.py", line 153, in main
    actionDelete(args)
  File "/home/runner/work/spiceai/spiceai/./.github/scripts/github_release.py", line 140, in actionDelete
    for asset in releaseInfo["assets"]:
                 ~~~~~~~~~~~^^^^^^^^^^
KeyError: 'assets'
```

Note: all steps succeeded, after manual retry one by one.

## To Reproduce

_Steps to reproduce the behavior:_

2. Click on '...'
3. Scroll down to '...'

## Expected behavior

_A clear and concise description of what you expected to happen._

### Spicepod

_Add the relevant `spicepod.yml` section._

### Output of `describe table`

_Add any relevant `describe table` output here._

### Output of `explain query`

_Add any relevant `explain query` output here._

E.g. `explain select 1`

```bash
+---------------+--------------------------------------+
| plan_type     | plan                                 |
+---------------+--------------------------------------+
| logical_plan  | Projection: Int64(1)                 |
|               |   EmptyRelation                      |
| physical_plan | ProjectionExec: expr=[1 as Int64(1)] |
|               |   PlaceholderRowExec                 |
|               |                                      |
+---------------+--------------------------------------+
```

### spice, spiced, OS info

- spice version: `spice version`
- spiced version: `spiced --version`
- OS info: `uname -a`

**Have you tried this on the latest `trunk` branch?**

- [ ] Yes
- [ ] No

**If possible, run `spiced` with DEBUG log level**
By setting the environment variable `SPICED_LOG="task_history=DEBUG,spiced=DEBUG,runtime=DEBUG,secrets=DEBUG,data_components=DEBUG,cache=DEBUG,extensions=DEBUG,spice_cloud=DEBUG"`

### PR DESCRIPTION
**Pull request by claudespice**
## Summary
Fixes #6566.

The release workflow occasionally fails mid-publish with
`KeyError: 'assets'` in `actionDelete`. Root cause: `getReleaseByTag`
only treated HTTP 404 as "release missing" and returned the raw JSON
for every other non-2xx status. When the GitHub API returns a transient
5xx, 403 rate limit, or similar, that response body is an error dict
(e.g. `{"message": "...", "documentation_url": "..."}`) without the
`assets`/`id`/`upload_url` keys callers depend on, so the next access
crashes the script and aborts the release.

## Changes
- `getReleaseByTag` now treats any non-2xx response (other than 404) as
  missing, logs the status and body, and returns `None`. Existing "no
  release" branches in `actionUpload` / `actionDelete` already handle
  that cleanly.
- `actionDelete` reads `assets` via `.get("assets", [])` so a release
  payload that lacks the key (e.g. a freshly created release) does not
  abort iteration with `KeyError`.
- New `.github/scripts/test_github_release.py` covers 200/404/5xx/403
  and non-JSON error bodies in `getReleaseByTag`, and exercises
  `actionDelete` against missing-release, missing-`assets`-key, empty
  assets, matching asset, and non-matching asset cases. The
  missing-`assets`-key test is the regression test for this bug — it
  fails on the original code.
- Small `release_script_test.yml` workflow runs the tests on PRs that
  touch the script (path-scoped, no other workloads triggered).

## Test plan
- [x] Added regression test for `KeyError: 'assets'`
- [x] Added edge case tests (5xx, 403, non-JSON body, empty assets,
      matching/non-matching asset names)
- [x] Verified regression test fails on the unfixed code (4 of 10 tests
      red without the patch)
- [x] All 10 tests pass with the fix
- [ ] CI green on PR

## Checklist
- [x] Root cause identified and fixed
- [x] Regression tests added
- [x] Edge cases covered
- [x] No snapshot deletions
- [x] No worktree paths or unrelated changes

### CODE REVIEW

File: .github/scripts/test_github_release.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+"""Unit tests for github_release.py."""
+import os
+import sys
+import unittest
```
[the remark this answers is not in the document]
Comment (claudespice) [reply .github/scripts/test_github_release.py:4]: Adopted a single-statement style: `import unittest.mock` plus references to `unittest.mock.patch(...)` / `unittest.mock.Mock(...)`. Note the bot suggestion as worded does not work — `import unittest` does not auto-import the `unittest.mock` submodule, so attribute access on `unittest.mock` raises `AttributeError`. Explicit `import unittest.mock` is required, which is what was done in 0083041b0. All 10 tests pass.

**Comment by claudespice** [comment]
@copilot review

**Comment by claudespice** [comment]
@copilot review please

**Comment by claudespice** [comment]
@copilot review

**Comment by claudespice** [comment]
@copilot please review

**Comment by claudespice** [comment]
@copilot review

**Review by lukekim — APPROVED** [review]
